core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -525,4 +526,20 @@ public void setCompressor(Compressor compressor) {}
       public void setDecompressor(Decompressor decompressor) {}
     }
   }
+
+  /**
+   * Returns a new status with the same code and description, but stripped of any other information
+   * (i.e. cause).
+   *
+   * <p>This is, so that the InProcess transport behaves in the same way as the other transports,
+   * when exchanging statuses between client and server and vice versa.
+   */
+  private static Status stripCause(Status status) {
+    if (status == null) {
+      return null;
+    }
+    return Status
+      .fromCodeValue(status.getCode().value())
```

Comment:
I like the idea of re-creating it with only the fields we want. That mirrors what we do in the other transports, and it means any additional fields to Status won't be propagated by default.

It also creates a new Status instance. I'm not sure whether that is a pro or a con.